### PR TITLE
[CI] Set up `test` workflow

### LINKED ISSUE #15: Have tests run in CI
We have (some) tests, let's have them run so we feel good. And then let's add more tests as part of PRs that touch the code base

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #15 

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Test with pytest
```
Comment (alyssadai) [inline .github/workflows/test.yaml:28]: 🍒 

```suggestion
        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi

    - name: Test with pytest
```

**Review by alyssadai — APPROVED** [review]
One 🍒 , otherwise 🧑‍🍳 !

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (surchs) [inline .github/workflows/test.yaml:27]: :cherries: why would we expect that there ever isn't a `requirements.txt`?

**Review by surchs — APPROVED** [review]
approved with one :cherries: to look at 

:cook:

**Comment by surchs** [comment]
ah, in case https://github.com/neurobagel/federation-api/issues/19 is simple, mind taking a look at that too?

File: .github/workflows/test.yaml (hunk 2 above)
Comment (rmanaem) [reply .github/workflows/test.yaml:27]: Copy pasta from the old script which itself was a copy pasta from *i think* their example. Will remove.

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.1.0`](https://github.com/neurobagel/federation-api/releases/tag/v0.1.0) :rocket:

**Pull request merged**